src/tools/rbd/action/Import.cc

```diff
@@ -853,18 +860,30 @@ static int do_import(librados::Rados &rados, librbd::RBD &rbd,
       goto done2;
     }
 
-    if ((fstat(fd, &stat_buf)) < 0) {
-      r = -errno;
-      std::cerr << "rbd: stat error " << path << std::endl;
-      goto done;
+    // fstat fails when used with Windows paths such as \\.\PhysicalDrive1.
+    // We'll rely on blkdev.get_size instead.
+    if (!utils::is_blk_dev(path)) {
+      if ((fstat(fd, &stat_buf)) < 0) {
+        r = -errno;
+        std::cerr << "rbd: stat error " << path << std::endl;
+        goto done;
+      }
+      if (S_ISDIR(stat_buf.st_mode)) {
+        r = -EISDIR;
+        std::cerr << "rbd: cannot import a directory" << std::endl;
+        goto done;
+      }
+      if (stat_buf.st_size)
+        size = (uint64_t)stat_buf.st_size;
```

Comment:
... and now that we have a dedicated branch for block devices, it would make sense to move `blkdev.get_size()` call there IMO:
```
      if (S_ISDIR(stat_buf.st_mode)) {
        r = -EISDIR;
        std::cerr << "rbd: cannot import a directory" << std::endl;
        goto done;
      }
      size = (uint64_t)stat_buf.st_size;
    } else {
#ifdef _WIN32
      if (import_format != 1) {
        std::cerr << "rbd: importing from raw block devices is only allowed "
                  << "using the v1 image format" << std::endl;
        r = -EINVAL;
        goto done;
      }
#endif
      int64_t bdev_size = 0;
      BlkDev blkdev(fd);
      r = blkdev.get_size(&bdev_size);
      if (r < 0) {
        std::cerr << "rbd: unable to get size of file/block device"
                  << std::endl;
        goto done;
      }
      ceph_assert(bdev_size >= 0);
      size = (uint64_t) bdev_size;
    }
#ifdef HAVE_POSIX_FADVISE
    posix_fadvise(fd, 0, 0, POSIX_FADV_SEQUENTIAL);
#endif
```